R/backend-postgres.R

```diff
@@ -96,7 +123,24 @@ sql_translate_env.PostgreSQLConnection <- function(con) {
           stop("Unrecognized arguments to `wday`", call. = FALSE)
         }
       },
-      yday = function(x) sql_expr(EXTRACT(DOY %FROM% !!x))
+      yday = function(x) sql_expr(EXTRACT(DOY %FROM% !!x)),
+      floor_date = function(x, unit = "seconds", week_start = NULL) {
+        supported_intervals <- c("second", "minute", "hour", "day", "week",
+                                 "month", "quarter", "year")
+
+        if (!unit %in% supported_intervals) {
+          stop(paste("Error: Invalid period name:", unit))
+        }
+
+        if (unit == "week" & !is.null(week_start)) {
```

Comment:
I think you should ignore this wrinkle of lubridate; the intent of the sql translations is convey the basic intention not absolutely every detail.